amp-beta-tester.php

```diff
@@ -0,0 +1,204 @@
+		return;
+	}
+	delete_site_transient( 'update_plugins' );
+}
+
+/**
+ * Hook into WP.
+ *
+ * @return void
+ */
+function init() {
+	// Abort init if AMP plugin is not active.
+	if ( ! defined( 'AMP__FILE__' ) ) {
+		add_action( 'admin_notices', __NAMESPACE__ . '\show_amp_not_active_notice' );
+		return;
+	}
+
+	add_filter( 'pre_set_site_transient_update_plugins', __NAMESPACE__ . '\update_amp_manifest' );
+	add_action( 'after_plugin_row_' . AMP_PLUGIN_FILE, __NAMESPACE__ . '\replace_view_version_details_link', 10, 2 );
+	add_action( 'admin_bar_menu', __NAMESPACE__ . '\show_unstable_reminder' );
+}
+
+/**
+ * Display an admin notice if the AMP plugin is not active.
+ *
+ * @return void
+ */
+function show_amp_not_active_notice() {
+	$error = esc_html__( 'AMP Beta Tester requires AMP to be active.', 'amp-beta-tester' );
+
+	// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+	echo "<div class='notice notice-error'><p><strong>{$error}</strong></p></div>";
+}
+
+/**
+ * Modifies the AMP plugin manifest to point to a new beta update if one exists.
+ *
+ * @param \stdClass $updates Object containing information on plugin updates.
+ * @return \stdClass
+ */
+function update_amp_manifest( $updates ) {
+	if ( ! isset( $updates->no_update ) ) {
+		return $updates;
+	}
+
+	$amp_zip_file = 'amp.zip';
+	$amp_manifest = isset( $updates->response[ AMP_PLUGIN_FILE ] )
+		? $updates->response[ AMP_PLUGIN_FILE ]
+		: $updates->no_update[ AMP_PLUGIN_FILE ];
+
+	$github_releases = get_amp_github_releases();
+
+	if ( is_array( $github_releases ) ) {
+		$amp_version = get_plugin_data( WP_PLUGIN_DIR . '/' . AMP_PLUGIN_FILE )['Version'];
+		$amp_updated = false;
+
+		foreach ( $github_releases as $release ) {
+			if ( $release->prerelease ) {
+				$release_version = $release->tag_name;
+
+				// If there is a new release, let's see if there is a zip available for download.
+				if ( version_compare( $release_version, $amp_version, '>' ) ) {
+					foreach ( $release->assets as $asset ) {
+						if ( $amp_zip_file === $asset->name ) {
+							$amp_manifest->new_version = $release_version;
+							$amp_manifest->package     = $asset->browser_download_url;
+							$amp_manifest->url         = $release->html_url;
+
+							// Set the AMP plugin to be updated.
+							$updates->response[ AMP_PLUGIN_FILE ] = $amp_manifest;
+							unset( $updates->no_update[ AMP_PLUGIN_FILE ] );
+
+							$amp_updated = true;
+							break;
+						}
+					}
+
+					if ( $amp_updated ) {
+						break;
+					}
+				}
+			}
+		}
+	}
+
+	return $updates;
+}
+
+/**
+ * Fetch AMP releases from GitHub.
+ *
+ * @return array|null
+ */
+function get_amp_github_releases() {
+	$raw_response = wp_remote_get( 'https://api.github.com/repos/ampproject/amp-wp/releases' );
+	if ( is_wp_error( $raw_response ) ) {
+		return null;
+	}
+	return json_decode( $raw_response['body'] );
+}
+
+/**
+ * Replace the 'View version details' link with the link to the release on GitHub.
```

Comment:
Might wanna look at how [Site Kit](https://github.com/google/site-kit-wp) is doing this.

Reply:
The plugin now hooks into the WP plugins API to show the correct version and also updates the download URL.